### PR TITLE
HDDS-5168. Use ReplicationConfig in OmKeyArgs

### PR DESCRIPTION
**Pull request by elek**
## What changes were proposed in this pull request?

During the implementation of HDDS-5145 I realized that OmKeyArgs also uses factor/type, it seems to be easier to convert it to replicationConfig as it's an in-memory class not a protobuf which is required to be persisted.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-5168

## How was this patch tested?

Full CI test.

### CODE REVIEW

File: hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/OzoneMultipartUploadPartListParts.java [hunk 1]
```diff
@@ -77,8 +79,16 @@ public boolean isTruncated() {
...
+    return ReplicationType
+            .fromProto(replicationConfig.getReplicationType());
+  }
+
+  @Deprecated
   public ReplicationFactor getReplicationFactor() {
-    return replicationFactor;
+    return ReplicationFactor
+            .fromProto(ReplicationConfig.getLegacyFactor(replicationConfig));
   }
```
Comment (sodonnel) [inline hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/OzoneMultipartUploadPartListParts.java:92]: Maybe we should add a `getReplicationConfig()` here, as we are deprecating the getType and Factor methods?

**Review by sodonnel — APPROVED** [review]
These changes LGTM. I had just one comment on a missing getter. It would be good to add it, but its not essential right now.

**Review by umamaheswararao — APPROVED** [review]

**Comment by umamaheswararao** [comment]
Changes looks good to me. I see. I think currently deprecated APIs used in the place where we may want to use getReplicationConfig API. I think thats ok for now? they are existing calls already( to deprecated APIs) and eventually new usage required to add that API(getReplicationConfig) and use it. If we wanted to add now, thats ok too. Thanks

File: hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/OzoneMultipartUploadPartListParts.java (hunk 1 above)
Comment (elek) [reply hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/OzoneMultipartUploadPartListParts.java:92]: Sure, I added the getter. It's not yet used, but can be useful for users of the Java API.

**Comment by elek** [comment]
Thanks the review @sodonnel and @umamaheswararao 

>  I think currently deprecated APIs used in the place where we may want to use getReplicationConfig API.

@umamaheswararao Can you please add some examples: I see examples where `S3StorageType` is used (which should be extended anyway), but not sure what other usages are in your mind...

**Pull request merged**